boards/xtensa/m5stickc_plus/m5stickc_plus.dts

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2022 AVSystem Sławomir Wolf Sp.j. (AVSystem)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/dts-v1/;
+
+#include <espressif/esp32.dtsi>
+#include "m5stickc_plus-pinctrl.dtsi"
+
+/ {
+	model = "esp32";
+	compatible = "espressif,esp32";
+
+	aliases {
+		led0 = &red_led;
+		sw0 = &user_button_0;
+		sw1 = &user_button_1;
+		uart-0 = &uart0;
+		i2c-0 = &i2c0;
+		watchdog0 = &wdt0;
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,flash = &flash0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		red_led: led_0 {
+			gpios =  <&gpio0 10 GPIO_ACTIVE_HIGH>;
+			label = "Red - LED0";
+		};
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+
+		user_button_0: button_0 {
+			label = "User button 0";
+			gpios = <&gpio1 5 GPIO_ACTIVE_LOW>;
+		};
+		user_button_1: button_1 {
+			label = "User button 1";
+			gpios = <&gpio1 7 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&cpu0 {
+	clock-frequency = <ESP32_CLK_CPU_240M>;
+};
+
+&cpu1 {
+	clock-frequency = <ESP32_CLK_CPU_240M>;
+};
+
+&uart0 {
+	status = "okay";
+	current-speed = <115200>;
+	pinctrl-0 = <&uart0_tx_gpio1 &uart0_rx_gpio3>;
+	pinctrl-names = "default";
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+/* IMU MPU-6886, RTC BM8563, PMU AXP192 */
+&i2c0 {
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+	sda-gpios = <&gpio0 21 GPIO_OPEN_DRAIN>;
+	scl-gpios = <&gpio0 22 GPIO_OPEN_DRAIN>;
+	pinctrl-0 = <&i2c0_sda_gpio21 &i2c0_scl_gpio22>;
```

Comment:
you could also use `&i2c0_default` from `esp32-pinctrl.dtsi` as it is using same pins